ec/src/lib.rs

```diff
@@ -274,10 +274,16 @@ pub trait AffineCurve:
     /// random group elements from a hash-function or RNG output.
     fn from_random_bytes(bytes: &[u8]) -> Option<Self>;
 
+    /// Multiplies `self` by the scalar represented by `bits`. `bits` must be a
+    /// big-endian bit-wise decomposition of the scalar.
+    fn mul_bits(&self, bits: impl Iterator<Item = bool>) -> Self::Projective;
+
     /// Performs scalar multiplication of this element with mixed addition.
     #[must_use]
-    fn mul<S: Into<<Self::ScalarField as PrimeField>::BigInt>>(&self, other: S)
-        -> Self::Projective;
+    #[inline]
+    fn mul<S: Into<<Self::ScalarField as PrimeField>::BigInt>>(&self, by: S) -> Self::Projective {
+        self.mul_bits(BitIteratorBE::new(by.into()))
```

Comment:
```suggestion
        self.mul_bits(BitIteratorBE::without_leading_zeros(by.into()))
```

Reply:
Is this more efficient? Why isn't it the default `new` implementation?

Reply:
It doesn't seem that this does anything in this case, because `mul_bits()` calls `skip_while()` anyway